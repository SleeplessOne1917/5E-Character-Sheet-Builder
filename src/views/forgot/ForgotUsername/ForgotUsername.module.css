.content {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.big-text {
    font-family: var(--font-fantasy);
    color: var(--davy);
}

.form {
    display: flex;
    flex-direction: column;
    align-items: center;
    height: 30%;
    font-family: var(--font-display);
}

.input-and-error-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
}

.input-and-label-container {
    display: flex;
    position: relative;
    flex-direction: column;
    align-items: center;
    margin-top: 1rem;
}

.input-container {
    padding-top: 1.2rem;
    display: flex;
    align-items: center;
}

.input {
    font-family: inherit;
    font-size: 1.1rem;
    background-color: var(--heaven-light);
    border: 1px solid var(--davy);
    border-radius: 0.5rem;
    color: var(--davy);
}

.input:focus,
.input:active {
    outline: var(--focus-outline);
}

.input::placeholder {
    color: transparent;
}

.label {
    position: absolute;
    bottom: 0.25rem;
    left: 0.5rem;
    color: #777;
    cursor: text;
    pointer-events: none;
}

.label-selected {
    color: var(--davy) !important;
    bottom: 0 !important;
    left: 0 !important;
    top: 0;
    font-size: 0.9rem !important;
    pointer-events: all !important;
}

.error {
    color: var(--wine-red);
    width: 90%;
    display: flex;
    justify-content: center;
    padding: 0.5rem;
    font-size: 0.8rem;
}

.input-error {
    border: 3px solid var(--error-red) !important;
}

.blurb {
    color: var(--davy);
    font-size: 0.8rem;
    padding: 0.5rem 4rem;
    margin: 0;
}

@media screen and (min-width: 768px) {
    .input {
        font-size: 1.6rem
    }

    .big-text {
        font-size: 2.5rem;
    }

    .label {
        font-size: 1.6rem;
    }

    .label-selected {
        font-size: 1.2rem !important;
    }

    .input-container {
        padding-top: 1.5rem
    }

    .error {
        font-size: 1rem;
    }

    .blurb {
        font-size: 1rem;
    }

}

@media screen and (min-width: 1024px) {
    .input {
        font-size: 2rem;
    }

    .label {
        font-size: 2rem;
    }

    .big-text {
        font-size: 3rem;
    }

    .input-container {
        padding-top: 2rem;
    }

    .label-selected {
        font-size: 1.6rem !important;
    }

    .error {
        font-size: 1.5rem;
    }
}

@media screen and (min-width: 1536px) {
    .blurb {
        font-size: 1.3rem;
    }
}