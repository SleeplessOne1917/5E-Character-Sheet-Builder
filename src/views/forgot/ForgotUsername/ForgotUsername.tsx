import Button, { ButtonType } from '../../../components/Button/Button';

import { Formik } from 'formik';
import MainContent from '../../../components/MainContent/MainContent';
import { ToastType } from '../../../types/toast';
import classes from './ForgotUsername.module.css';
import forgotUsernameSchema from '../../../yup-schemas/forgotUsernameSchema';
import { show } from '../../../redux/features/toast';
import { useAppDispatch } from '../../../hooks/reduxHooks';

const ForgotUsername = () => {
	const dispatch = useAppDispatch();

	return (
		<MainContent testId="forgot-username">
			<div className={classes.content}>
				<h1 className={classes['big-text']}>Forgot username</h1>
				<p className={classes.blurb}>
					Enter your email address. If it&apos;s in the system, you&apos;ll
					receive an email with a link to view your username.
				</p>
				<Formik
					initialValues={{ email: '' }}
					validationSchema={forgotUsernameSchema}
					onSubmit={({}, { resetForm }) => {
						dispatch(
							show({
								closeTimeoutSeconds: 10,
								message: 'Email sent!',
								type: ToastType.success
							})
						);
						resetForm();
					}}
				>
					{({
						touched,
						errors,
						values,
						handleChange,
						handleBlur,
						handleSubmit,
						isSubmitting
					}) => (
						<form onSubmit={handleSubmit} className={classes.form}>
							<div className={classes['input-and-error-container']}>
								<div className={classes['input-and-label-container']}>
									<div className={classes['input-container']}>
										<input
											className={`${classes.input}${
												touched.email && errors.email
													? ` ${classes['input-error']}`
													: ''
											}`}
											id="email"
											name="email"
											type="text"
											value={values.email}
											placeholder="Email"
											onChange={handleChange}
											onBlur={handleBlur}
										/>{' '}
									</div>
									<label
										htmlFor="email"
										className={`${classes.label}${
											values.email && values.email.length > 0
												? ` ${classes['label-selected']}`
												: ''
										}`}
									>
										Email
									</label>
								</div>
								{touched.email && errors.email && (
									<div className={classes.error}>{errors.email}</div>
								)}
							</div>
							<Button
								disabled={isSubmitting}
								type={ButtonType.submit}
								positive
								size="medium"
								spacing={4}
							>
								Send username reminder
							</Button>
						</form>
					)}
				</Formik>
			</div>
		</MainContent>
	);
};

export default ForgotUsername;
