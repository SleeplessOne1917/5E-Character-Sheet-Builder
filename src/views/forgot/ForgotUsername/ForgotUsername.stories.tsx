import { ComponentMeta, ComponentStory } from '@storybook/react';

import ForgotUsername from './ForgotUsername';
import { Provider } from 'react-redux';
import { getTestStore } from '../../../redux/store';

export default {
	title: 'Views/ForgotUsername',
	component: ForgotUsername,
	decorators: [story => <Provider store={getTestStore()}>{story()}</Provider>]
} as ComponentMeta<typeof ForgotUsername>;

const Template: ComponentStory<typeof ForgotUsername> = (args: any) => (
	<ForgotUsername {...args} />
);

export const Default = Template.bind({});
