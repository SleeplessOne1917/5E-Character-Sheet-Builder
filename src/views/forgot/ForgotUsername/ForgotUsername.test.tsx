import * as stories from './ForgotUsername.stories';

import { render, screen } from '@testing-library/react';

import { composeStories } from '@storybook/testing-react';

const { Default } = composeStories(stories);

it('renders correctly', () => {
	render(<Default />);

	expect(screen.getByTestId('forgot-username')).toMatchSnapshot();
});
