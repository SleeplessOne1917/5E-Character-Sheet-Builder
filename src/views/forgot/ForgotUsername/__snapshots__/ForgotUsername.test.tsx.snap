// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  class="main"
  data-testid="forgot-username"
>
  <div
    class="content"
  >
    <div
      class="content"
    >
      <h1
        class="big-text"
      >
        Forgot username
      </h1>
      <p
        class="blurb"
      >
        Enter your email address. If it's in the system, you'll receive an email with a link to view your username.
      </p>
      <form
        class="form"
      >
        <div
          class="input-and-error-container"
        >
          <div
            class="input-and-label-container"
          >
            <div
              class="input-container"
            >
              <input
                class="input"
                id="email"
                name="email"
                placeholder="Email"
                type="text"
                value=""
              />
               
            </div>
            <label
              class="label"
              for="email"
            >
              Email
            </label>
          </div>
        </div>
        <button
          class="button positive"
          style="font-size: 1.3rem; padding: 1rem; margin: 1rem;"
          type="submit"
        >
          Send username reminder
        </button>
      </form>
    </div>
  </div>
</main>
`;
